### PR TITLE
refresh horovod

### PR DESCRIPTION
**Pull request by cdeepali**
## Checklist before submitting

- [ ] Did you read the [contributor guide](https://github.com/open-ce/open-ce/blob/main/CONTRIBUTING.md)?
- [ ] Did you update any affected [documentation](https://github.com/open-ce/open-ce/blob/main/doc/)?
- [ ] Did you write any tests to validate this change?

## Description

Fixes # (issue).

## Review process to land 

1. All tests and other checks must succeed.
2. At least one [maintainer](https://github.com/open-ce/open-ce/blob/main/MAINTAINERS.md) must review and approve.
3. If any  [maintainer](https://github.com/open-ce/open-ce/blob/main/MAINTAINERS.md) requests changes, they must be addressed.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {% set name = "horovod" %}
-{% set version = "0.24.2" %}
+{% set version = "0.24.3" %}
```
Comment (alexlang74) [inline recipe/meta.yaml:2]: Shouldn't this be `0.25` ?

Comment (cdeepali) [reply recipe/meta.yaml]: Yes it will be 0.25. I am yet to verify 0.25 locally, once done I will push that change.

**Review by alexlang74 — APPROVED** [review]

**Comment by npanpaliya** [comment]
We still need to keep conda_build_config file to have gcc 8.2 for cuda 11 builds. Right?
Also build-config.yaml was kept over here to have that explicit check for `if build_type ==cuda` so the recipe is built only for cuda variant. I think we should keep this file as well.

**Comment by cdeepali** [comment]
@npanpaliya Accidentally deleted those files. Will push them back.

**Review by npanpaliya — APPROVED** [review]

**Pull request merged**